### ISSUE TITLE
hasComponent might return undefined when the component is not registered yet

### ISSUE THREAD

**Issue by gepz** [NateTheGreatt/bitECS#49]
Is this the expected behavior?
Incidentally, based on [this comment](https://github.com/NateTheGreatt/bitECS/issues/11#issuecomment-815149173) is hasComponent not recommended to use?
If so, is there another way to reuse the queried results to further filter the entities with specific tags without requerying?

**Comment by NateTheGreatt** [comment]
good catch, i think it should instead return `false` in that case. this change will be in the next release.

also `hasComponent` is fine to use! it depends on the use case i suppose... one could trade these `hasComponent` checks for more specific queries, e.g.
```js
const queryA = defineQuery([A])
const systemA = world => {
  for (const eid of queryA(world)) {
    doStuff(eid)
    if(hasComponent(world, B, eid)) {
      doOtherStuffB(eid)
    }
  }
}
```
could be turned into this:
```js
const queryA = defineQuery([A])
const systemA = world => {
  for (const eid of queryA(world)) {
    doStuffA(eid)
  }
}
const queryB = defineQuery([B])
const systemB = world => {
  for (const eid of queryB(world)) {
    doOtherStuffB(eid)
  }
}
```
or vice versa.

whatever works best for your needs :+1:

**Comment by NateTheGreatt** [comment]
closing out, published in `0.3.21-5`
